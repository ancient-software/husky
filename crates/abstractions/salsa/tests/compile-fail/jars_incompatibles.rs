#[salsa::jar]
struct JarWithRetRef(MyInput);

#[salsa::jar]
struct JarWithDb(MyInput);

#[salsa::jar]
struct JarWithNoEq(MyInput);

#[salsa::jar]
struct JarWithJar(MyInput);

#[salsa::jar]
struct JarWithData(MyInput);

#[salsa::jar]
struct JarWithRecover(MyInput);

#[salsa::jar]
struct JarWithLru(MyInput);

#[salsa::jar]
struct JarWithConstructor(MyInput);

#[salsa::input(jar = Jar1)]
struct MyInput {
    field: u32,
}

fn main() {}
