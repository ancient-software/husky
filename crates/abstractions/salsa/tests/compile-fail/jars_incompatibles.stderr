error: `return_ref` option not allowed here
 --> tests/compile-fail/jars_incompatibles.rs:1:23
  |
1 | #[salsa::jar]
  |                       ^^^^^^^^^^

error: `specify` option not allowed here
 --> tests/compile-fail/jars_incompatibles.rs:4:23
  |
4 | #[salsa::jar]
  |                       ^^^^^^^

error: `no_eq` option not allowed here
 --> tests/compile-fail/jars_incompatibles.rs:8:23
  |
8 | #[salsa::jar]
  |                       ^^^^^

error: `jar` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:11:23
   |
11 | #[salsa::jar]
   |                       ^^^

error: `data` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:14:23
   |
14 | #[salsa::jar]
   |                       ^^^^

error: `recovery_fn` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:17:23
   |
17 | #[salsa::jar]
   |                       ^^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:20:23
   |
20 | #[salsa::jar]
   |                       ^^^

error: `constructor` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:23:23
   |
23 | #[salsa::jar]
   |                       ^^^^^^^^^^^

error[E0412]: cannot find type `Jar1` in this scope
  --> tests/compile-fail/jars_incompatibles.rs:26:22
   |
26 | #[salsa::input(jar = Jar1)]
   |                      ^^^^ not found in this scope

error[E0412]: cannot find type `Jar1` in this scope
  --> tests/compile-fail/jars_incompatibles.rs:26:22
   |
26 | #[salsa::input(jar = Jar1)]
   |                      ^^^^ not found in this scope
27 | struct MyInput {
   |               - help: you might be missing a type parameter: `<Jar1>`
